test/modules/post/test/smb.rb

```diff
@@ -0,0 +1,202 @@
+        'License' => MSF_LICENSE,
+        'Author' => [ 'sjanusz-r7'],
+        'Platform' => all_platforms,
+        'SessionTypes' => [ 'smb' ]
+      )
+    )
+  end
+
+  def test_console_help
+    it 'should support the help command' do
+      stdout = with_mocked_console(session) { |console| console.run_single('help') }
+      ret = true
+      ret &&= stdout.buf.include?('Core Commands')
+      ret &&= stdout.buf.include?('Shares Commands')
+      ret &&= stdout.buf.include?('Local File System Commands')
+      ret
+    end
+  end
+
+  def test_upload_and_download
+    readonly_share = 'readonly'
+    modifiable_share = 'modifiable'
+
+    Tempfile.create do |temp_file|
+      filename = File.basename(temp_file)
+      full_path = temp_file.to_path
+
+      it 'should support uploading files' do
+        stdout = with_mocked_console(session) do |console|
+          console.run_single("shares -i #{modifiable_share}")
+          console.run_single("upload #{full_path} #{filename}")
+        end
+
+        ret = true
+        # Or filename?
+        ret &&= stdout.buf.include?("#{full_path} uploaded to #{filename}")
+        ret
+      end
+
+      it 'should not upload to readonly share' do
+        stdout = with_mocked_console(session) do |console|
+          console.run_single("shares -i #{readonly_share}")
+          console.run_single("upload #{full_path} #{filename}")
+        end
+
+        ret = true
+        ret &&= stdout.buf.include?('Error running command upload')
+        ret &&= stdout.buf.include?('The server responded with an unexpected status code: STATUS_ACCESS_DENIED')
+        ret
+      end
+
+      it 'should support deleting files' do
+        stdout = with_mocked_console(session) do |console|
+          console.run_single("shares -i #{modifiable_share}")
+          console.run_single("delete #{filename}")
+        end
+
+        ret = true
+        ret &&= stdout.buf.include?("Deleted #{filename}")
+        ret
+      end
+    end
+
+    Tempfile.create do |temp_file|
+      remote_filename = 'hello_world.txt'
+      remote_dir = 'text_files'
+      full_path = temp_file.to_path
+
+      it 'should support downloading files' do
+        stdout = with_mocked_console(session) do |console|
+          console.run_single("shares -i #{modifiable_share}")
+          console.run_single("cd #{remote_dir}")
+          console.run_single("download #{remote_filename} #{full_path}")
+        end
+
+        ret = true
+        ret &&= stdout.buf.include?("Downloaded #{remote_dir}\\#{remote_filename} to #{full_path}")
+        ret
+      end
+    end
+  end
+
+  def test_files
+    modifiable_share = 'modifiable'
+
+    it 'should output files in the current directory' do
+      stdout = with_mocked_console(session) do |console|
+        console.run_single("shares -i #{modifiable_share}")
+        console.run_single('ls')
+      end
+
+      ret = true
+      ret &&= stdout.buf.include?('recursive')
```

Comment:
Not a blocker; these tests will be quite rigid - I wouldn't be able to run these tests against an arbitrary server for instance